buffer/src/main/java/io/netty/buffer/ByteBufUtil.java

```diff
@@ -565,10 +565,25 @@ static ByteBuf encodeString0(ByteBufAllocator alloc, boolean enforceHeap, CharBu
         }
     }
 
+    static String readUtf8(AbstractByteBuf buf, int readerIndex, int length) { // Fast-Path for a AbstractByteBuf
+        buf.checkIndex(readerIndex, length);
+        UTF8Decoder.UTF8Processor processor = new UTF8Decoder.UTF8Processor(new char[length]);
```

Comment:
My2cents: Instead of allocating UTF8Processor and char[] over and over again, you could use a `StringBuilder`, cache your instances in a ThreadLocal (maybe only up to a given limit) and recycle them.